File: Java/Insert-Node-in-a-Binary-Search-Tree.java
/**
 * Definition of TreeNode:
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left, right;
 *     public TreeNode(int val) {
 *         this.val = val;
 *         this.left = this.right = null;
 *     }
 * }
 */
public class Solution {
    /**
     * @param root: The root of the binary search tree.
     * @param node: insert this node into the binary search tree
     * @return: The root of the new binary search tree.
     */
    public TreeNode insertNode(TreeNode root, TreeNode node) {
        // write your code here
        if (root == null) return node;
        TreeNode cur = root;
        while (cur != null) {
        	if (node.val == cur.val) return root;
        	if (node.val > cur.val) {
        		if (cur.right == null) {
        			cur.right = node;
        			return root;
        		} else {
        			cur = cur.right;
        		}
        	} else {
        		if (cur.left == null) {
        			cur.left = node;
        			return root;
        		} else {
        			cur = cur.left;
        		}
        	}
        }
        return root;
    }
}

File: Python/Insert-Node-in-a-Binary-Search-Tree.py
"""
Definition of TreeNode:
class TreeNode:
    def __init__(self, val):
        self.val = val
        self.left, self.right = None, None
"""
class Solution:
    """
    @param root: The root of the binary search tree.
    @param node: insert this node into the binary search tree.
    @return: The root of the new binary search tree.
    """
    def insertNode(self, root, node):
        # write your code here
        if root is None:
            return node
        cur = root
        while cur is not None:
            if node.val == cur.val:
                return root
            elif cur.val < node.val:
                if cur.right is None:
                    cur.right = node
                    return root
                else:
                    cur = cur.right
            elif cur.val > node.val:
                if cur.left is None:
                    cur.left = node
                    return root
                else:
                    cur = cur.left
        return root